Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -20,56 +20,54 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-/// A queue that executes asynchronous tasks enqueued from a nonisolated context.
+/// A queue that enables enqueing ordered asynchronous tasks from a nonisolated context onto an adopted actor's isolated context.
 /// Tasks are guaranteed to begin executing in the order in which they are enqueued. However, if a task suspends it will allow subsequently enqueued tasks to begin executing.
-/// Asynchronous tasks sent to this queue execute as they would in an `actor` type, allowing for re-entrancy and non-FIFO behavior when an individual task suspends.
+/// This queue exhibits the execution behavior of an actor: tasks sent to this queue can re-enter the queue, and tasks may execute in non-FIFO order when a task suspends.
 ///
-/// An `ActorQueue` is used to ensure tasks sent from a nonisolated context to a single `actor`'s isolated context begin execution in order.
-/// Here is an example of how an `ActorQueue` should be utilized within an `actor`:
+/// An `ActorQueue` ensures tasks sent from a nonisolated context to a single actor's isolated context begin execution in order.
+/// Here is an example of how an `ActorQueue` should be utilized within an actor:
 /// ```swift
 /// public actor LogStore {
 ///
+///     public init() {
+///         queue.adoptExecutionContext(of: self)
+///     }
+///
 ///     nonisolated
 ///     public func log(_ message: String) {
-///         queue.async {
-///             await self.append(message)
+///         queue.async { myself in
+///             myself.logs.append(message)
 ///         }
 ///     }
 ///
 ///     nonisolated
 ///     public func retrieveLogs() async -> [String] {
-///         await queue.await { await self.logs }
-///     }
-///
-///     private func append(_ message: String) {
-///         logs.append(message)
+///         await queue.await { myself in myself.logs }
 ///     }
 ///
-///     private let queue = ActorQueue()
+///     private let queue = ActorQueue<LogStore>()
 ///     private var logs = [String]()
 /// }
 /// ```
 ///
-/// - Warning: Execution order is not guaranteed unless the enqueued tasks interact with a single `actor` instance.
-public final class ActorQueue {
+/// - Warning: The lifecycle of an `ActorQueue` should not exceed that of the adopted actor.
+public final class ActorQueue<ActorType: Actor> {
 
     // MARK: Initialization
 
     /// Instantiates an actor queue.
-    /// - Parameter priority: The baseline priority of the tasks added to the asynchronous queue.
-    public init(priority: TaskPriority? = nil) {
```

Comment:
Why do we still allow a `FIFOQueue` to be initialized with a `priority` but not an actor queue? Does it have something to do with how we're using the actor's context for task execution?

Reply:
> Does it have something to do with how we're using the actor's context for task execution?

Exactly this. We could create a high-priority for-loop but we're always going to be waiting on the actor's context anyways so it seems not terribly useful to have a `priority` argument here. We can always add it back.